pkg/roachprod/install/cockroach.go

```diff
@@ -434,6 +438,16 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 		startOpts.AdminUIPort = config.DefaultAdminUIPort
 	}
 
+	for _, hook := range startOpts.PreStartHooks {
+		if hook.Fn == nil {
+			continue
+		}
+		l.Printf("running pre-start hook: %s", hook.Name)
+		if err := hook.Fn(ctx); err != nil {
```

Comment:
We have to be careful in case of (user) panic. The mixedversion framework does it via panicAsError.

Reply:
Technically, we should also impose a maximum running time for each hook, but seems fine for the first iteration.

Reply:
Done, good catch